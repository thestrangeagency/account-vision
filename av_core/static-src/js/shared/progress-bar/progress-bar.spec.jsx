import React from 'react';
import { shallow } from 'enzyme';

import ProgressBar from './progress-bar';

test('it renders', () => {
  const wrapper = shallow(<ProgressBar />);
  expect(wrapper).toMatchSnapshot();

  wrapper.setProps({ progress: 0.5 });
  expect(wrapper).toMatchSnapshot();

  wrapper.setProps({ progress: 10 });
  expect(wrapper).toMatchSnapshot();

  wrapper.setProps({ progress: -1 });
  expect(wrapper).toMatchSnapshot();
});
