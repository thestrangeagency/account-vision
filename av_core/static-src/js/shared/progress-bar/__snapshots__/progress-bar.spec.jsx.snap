// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<div
  className="progress"
>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow={0}
    className="progress-bar"
    role="progressbar"
    style={
      Object {
        "width": "0%",
      }
    }
  />
</div>
`;

exports[`it renders 2`] = `
<div
  className="progress"
>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow={50}
    className="progress-bar"
    role="progressbar"
    style={
      Object {
        "width": "50%",
      }
    }
  />
</div>
`;

exports[`it renders 3`] = `
<div
  className="progress"
>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow={100}
    className="progress-bar"
    role="progressbar"
    style={
      Object {
        "width": "100%",
      }
    }
  />
</div>
`;

exports[`it renders 4`] = `
<div
  className="progress"
>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow={0}
    className="progress-bar"
    role="progressbar"
    style={
      Object {
        "width": "0%",
      }
    }
  />
</div>
`;
